app/src/main/java/com/example/healthinspector/Fragments/ScanFlow/ScanFragment.java

```diff
@@ -135,12 +138,25 @@ public void onResponse(JSONObject response) {
                                     JSONObject nutrientLevelsJSON = response.getJSONObject("product").getJSONObject("nutrient_levels");
                                     Iterator iterator =  nutrientLevelsJSON.keys();
                                     while (iterator.hasNext()){
-                                        nutrientLevels.add(nutrientLevelsJSON.getString(iterator.next().toString()));
+                                        String key = iterator.next().toString();
+                                        nutrientLevels.add(key + ": " + nutrientLevelsJSON.getString(key));
                                     }
+
+                                    //getting "additives_original_tags"
+
                                     //send to next screen and populate views
-                                    //creating a scannedProduct object for to wrap with
+                                    //creating a scannedProduct object to wrap with Parceler and put pas to the next fragment
                                     ScannedProduct scannedProduct = new ScannedProduct(productName, healthInspectorScore, ingredients, ingredientsAnalysis, novaGroup, nutrientLevels);
-                                    
+                                    //creating fragment transaction and passing the object into the bundle which is then passed into the fragment as an argument
+                                    FragmentTransaction fragmentTransaction =  getActivity().getSupportFragmentManager().beginTransaction();
+                                    fragmentTransaction.setTransition(FragmentTransaction.TRANSIT_FRAGMENT_OPEN);
+                                    ProductDetailsFragment productDetailsFragment = new ProductDetailsFragment();
+                                    Bundle bundle = new Bundle();
+                                    bundle.putParcelable("scannedProduct", Parcels.wrap(scannedProduct));
```

Comment:
store this "scannedProduct" in a constant so you can access it in both fragments. It is a good practice to have all raw strings like this defined in one place